### ISSUE TITLE
Unable to fetch instanceUuid from VM

### ISSUE THREAD

**Issue by manchiu** [fog/fog-vsphere#123]
Hi, 
ERROR MSG: undefined method 'instanceUuid' for nil:NilClass
When I launch a new VM in vCenter and call list_hosts at the same time, this error occurs.
For `vm_ids:          Proc.new { host[:vm].map {|vm| vm.config.instanceUuid } }` in line 30,  are there any workarounds if vm.config.instanceUuid cannot be found like rescue nil? See if you have any concern
Thanks!
Greetings
Alex

/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:17:in `block in host_attributes'
/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:17:in `map'
/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:17:in `host_attributes'
/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:7:in `block in list_hosts'
/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:7:in `map'
/usr/share/rvm/gems/ruby-2.1.2@app360/gems/fog-vsphere-1.7.0/lib/fog/vsphere/requests/compute/list_hosts.rb:7:in `list_hosts'

**Comment by timogoebel** [comment]
@manchiu: Thanks for the report. Basically, you can just `rescue nil`. But I'm wondering why this is happening in the first place.

Can you try with something like this and post the output?

```
def host_attributes(host, filters)
          {
            datacenter: filters[:datacenter],
            cluster:    filters[:cluster],
            name:       host[:name],
            vm_ids:     host[:vm].map {|vm| puts vm.class; vm.config.instanceUuid }
          }
        end
```

or use pry debugger to find out, why `vm.config` is `nil`.

**Comment by manchiu** [comment]
@timogoebel , I have tried and the output is as follow: 

VirtualMachine
VirtualMachine
VirtualMachine
VirtualMachine
VirtualMachine
[2018-03-01T16:54:22.471895 #11503] ERROR -- : [v4.3.dev] Appcara::Cloud::Vcenter::Host#rescue in list - caused by /home/alexchiu/appcara/appstack-cloudengine/vendor/fog/vsphere/requests/compute/list_hosts.rb:29: **undefined method `instanceUuid' for nil:NilClass
Did you mean?  instance_values
NoMethodError: undefined method `instanceUuid' for nil:NilClass**

I think probably the process of launching a new VM has not yet completed and at the same time we try to fetch the information of this VM. Therefore the error occurs.

This is how I reproduce the case stably:
1. Launch a new VM in vCenter
2. try to call list_hosts
3. Error occurs

**Comment by timogoebel** [comment]
@manchiu: I think a `rescue nil` approach is feasible here. Do you feel comfortable opening a PR?

**Comment by manchiu** [comment]
@timogoebel : Sure. I have just opened a PR. May you help to review?

**Comment by timogoebel** [comment]
Fixed via #126.
